private-templates-service/client/src/components/AllCards/AllCards.tsx

```diff
@@ -38,23 +48,79 @@ const mapDispatchToProps = (dispatch: any) => {
     },
     toggleView: (viewType: ViewType) => {
       dispatch(setViewToggleType(viewType));
+    },
+    getAllTags: () => {
+      dispatch(getAllTags());
+    },
+    addSelectedTag: (tag: string) => {
+      dispatch(addSelectedTag(tag))
+    },
+    removeSelectedTag: (tag: string) => {
+      dispatch(removeSelectedTag(tag))
+    },
+    clearSelectedTags: () => {
+      dispatch(clearSelectedTags());
     }
   };
 };
 interface Props extends RouteComponentProps {
-  templates: AllTemplateState;
   setPage: (currentPageTitle: string, currentPage: string) => void;
   setSearchBarVisible: (isSearchBarVisible: boolean) => void;
   toggleView: (viewType: ViewType) => void;
+  getAllTags: () => void;
+  addSelectedTag: (tag: string) => void;
+  removeSelectedTag: (tag: string) => void;
+  clearSelectedTags: () => void;
```

Comment:
are those intended for removing a single tag vs removing all of them? would it be better to have only one and if parameter is not given remove all of them?

Reply:
I can't come up with naming of the method that wouldn't make the purpose of the function confusing. If you have any idea, please let me know!

Reply:
The naming looks fine. I meant, if they are doing the same thing: i.e delete a selected tag or delete all tags, there could be a single method do both depending on the arguments.